xtc/src/ledger.rs

```diff
@@ -153,32 +152,31 @@ pub struct ErrorDetails {
     code: APIError,
 }
 
-static InsufficientBalanceError: ErrorDetails = ErrorDetails {
+const INSUFFICIENT_BALANCE_ERROR: ErrorDetails = ErrorDetails {
     msg: "Insufficient Balance",
     code: APIError::InsufficientBalance,
 };
 
-static InsufficientAllowanceError: ErrorDetails = ErrorDetails {
+const INSUFFICIENT_ALLOWANCE_ERROR: ErrorDetails = ErrorDetails {
     msg: "Insufficient Allowance",
     code: APIError::InsufficientAllowance,
 };
 
-static UnknownError: ErrorDetails = ErrorDetails {
+const UNKNOWN_ERROR: ErrorDetails = ErrorDetails {
     msg: "Unknown",
     code: APIError::Unknown,
 };
 
-#[query]
-fn totalSupply() -> Nat {
+#[query(name = "totalSupply")]
+fn total_supply() -> Nat {
```

Comment:
if a function has 
query or update attribute, it means those functions are part of our canister service interface, so we cannot change it before we talk to upstream. We also need to change the candid files whenever we change attributed functions. Could you please revert those back?

Reply:
I was told that using `name = "foo"` would rebind the interface?

Reply:
Just checked with the debug env var
```
--- IC_CDK_MACROS DEBUG ---
ic_query
    attr: name = "totalSupply"
    item: fn total_supply() -> Nat { StatsData :: get().supply }
--------- RESULT  ---------
#[export_name = "canister_query totalSupply"] fn total_supply_9_()
{
    ic_cdk :: setup() ; ic_cdk ::
    block_on(async
             {
                 let() = ic_cdk :: api :: call :: arg_data() ; let result =
                 total_supply() ; ic_cdk :: api :: call :: reply((result,))
             }) ;
} fn total_supply() -> Nat { StatsData :: get().supply }
---------------------------
```

that looks right?

Reply:
yeah that's correct.

Reply:
> Just checked with the debug env var
> 
> ```
> --- IC_CDK_MACROS DEBUG ---
> ic_query
>     attr: name = "totalSupply"
>     item: fn total_supply() -> Nat { StatsData :: get().supply }
> --------- RESULT  ---------
> #[export_name = "canister_query totalSupply"] fn total_supply_9_()
> {
>     ic_cdk :: setup() ; ic_cdk ::
>     block_on(async
>              {
>                  let() = ic_cdk :: api :: call :: arg_data() ; let result =
>                  total_supply() ; ic_cdk :: api :: call :: reply((result,))
>              }) ;
> } fn total_supply() -> Nat { StatsData :: get().supply }
> ---------------------------
> ```
> 
> that looks right?

sorry, didn't see you renamed it with the attribute